projects/rero/ng-core/src/lib/record/detail/detail.component.ts

```diff
@@ -95,18 +103,28 @@ export class DetailComponent implements OnInit {
       (record) => {
         this.record = record;
 
-        const config = this.recordUiService.getResourceConfig(this.route.snapshot.data.types, type);
+        this.recordUiService.types = this.route.snapshot.data.types;
 
-        if (config.canDelete) {
-          config.canDelete(this.record).subscribe((result: ActionStatus) => {
-            this.deleteStatus = result;
-          });
-        }
+        this.recordUiService.canReadRecord$(this.record, type).subscribe(result => {
+          if (result.can === false) {
+            this.toastrService.error(
+              _('You cannot read this record'),
+              _(type)
+            );
+            this.location.back();
+          }
+        });
+
+        this.recordUiService.canDeleteRecord$(this.record, type).subscribe(result => {
+          this.deleteStatus = result;
+        });
+
+        this.recordUiService.canUpdateRecord$(this.record, type).subscribe(result => {
+          this.updateStatus = result;
+        });
 
-        if (config.canUpdate) {
-          config.canUpdate(this.record).subscribe((result: ActionStatus) => {
-            this.updateStatus = result;
-          });
+        if (typeof this.route.snapshot.data.adminMode !== 'undefined') {
```

Comment:
`typeof`?

Reply:
You're right, as I read [here](https://basarat.gitbooks.io/typescript/docs/javascript/null-undefined.html), the best way to do this is : 

```typescript
if (this.route.snapshot.data.adminMode != null) {
    this.adminMode = this.route.snapshot.data.adminMode;
}
```

Reply:
done in other PR